kernel/src/main/java/org/kframework/compile/checks/CheckTokens.java

```diff
@@ -0,0 +1,43 @@
+package org.kframework.compile.checks;
+
+import com.google.common.collect.ImmutableSet;
+import org.kframework.definition.Module;
+import org.kframework.definition.Production;
+import org.kframework.definition.Sentence;
+import org.kframework.utils.errorsystem.KEMException;
+
+import java.util.Set;
+
+public class CheckTokens {
+    private final Set<KEMException> errors;
+    private final Module m;
+    private static final ImmutableSet<String> ignoredSortNames = ImmutableSet.of("Bag", "IOFile", "IOInt", "IOString", "K", "KBott", "KItem", "KLabel", "KList", "Sort");
+
+    public CheckTokens(Set<KEMException> errors, Module m) {
+        this.errors = errors;
+        this.m = m;
+    }
+
+    // This check ensures that sorts containing token declarations only contain syntax declarations that are also tokens (or macros).
+    public void check(Sentence sentence) {
```

Comment:
`private void check(Production p) {`
You can match directly on Production and do away with this check.

Reply:
ah, thanks that would make things more clear